firedrake/slate/slac/optimise.py

```diff
@@ -99,7 +100,7 @@ def _push_block_tensor(expr, self, indices):
 @_push_block.register(AssembledVector)
 def _push_block_assembled_vector(expr, self, indices):
     """Turns a Block on an AssembledVector into the  specialized node BlockAssembledVector."""
-    return BlockAssembledVector(expr._function, Block(expr, indices).form, indices) if indices else expr
+    return BlockAssembledVector(expr._function, Block(expr, indices), indices) if indices else expr
```

Comment:
By the way, can we change the interface so that `BlockAssembledVector` would take `expr` and `indices` as arguments?

Reply:
Yes done